### PR TITLE
Support synthetic join handles in the parser and recipe data structures

### PR DESCRIPTION
**Pull request by piotrswigon**

### CODE REVIEW

File: src/runtime/recipe/handle.ts [hunk 1]
```diff
@@ -376,4 +384,9 @@ export class Handle implements Comparable<Handle> {
   findConnectionByDirection(dir: Direction): HandleConnection|undefined {
     return this._connections.find(conn => conn.direction === dir);
   }
+
+  associateHandle(handle: Handle) {
```
Comment (Cypher1) [inline src/runtime/recipe/handle.ts]: Is there a way to assert ordering of associated handles?
My concern is that the tuple might accidentally get reordered.

File: src/runtime/tests/manifest-parser-test.ts [hunk 2]
```diff
@@ -57,6 +57,13 @@ describe('manifest parser', () => {
         h1: create 'my-id' #anotherTag @ttl(1h)
         h2: create @ttl ( 30m )`);
   });
+  it('parses recipes with a synthetic join handles', () => {
```
Comment (Cypher1) [inline src/runtime/tests/manifest-parser-test.ts]: Probably worth having a test that it throws when joining over non-handles

Comment (piotrswigon) [reply src/runtime/tests/manifest-parser-test.ts]: Added in manifest-test, as the parser itself does not try to understand what the joined identifiers mean.

Comment (Cypher1) [reply src/runtime/tests/manifest-parser-test.ts]: Thanks, perfect.

**Pull request merged**